workflows/fastmatchirida.nf

```diff
@@ -127,22 +131,7 @@ workflow FASTMATCH {
         exit 1, "--pd_columns ${params.pd_columns}: Does not exist but was passed to the pipeline. Exiting now."
     }
 
-    if(params.gm_thresholds == null || params.gm_thresholds == ""){
-        exit 1, "--gm_thresholds ${params.gm_thresholds}: Cannot pass null or empty string"
-    }
-
-    gm_thresholds_list = params.gm_thresholds.toString().split(',')
-    if (params.pd_distm == 'hamming') {
-        if (gm_thresholds_list.any { it != null && it.contains('.') }) {
-            exit 1, ("'--pd_distm ${params.pd_distm}' is set, but '--gm_thresholds ${params.gm_thresholds}' contains fractions."
-                    + " Please either set '--pd_distm scaled' or remove fractions from distance thresholds.")
-        }
-    } else if (params.pd_distm == 'scaled') {
-        if (gm_thresholds_list.any { it != null && (it as Float < 0.0 || it as Float > 100.0) }) {
```

Comment:
The purpose of these if-else statements in the gasclustering pipeline was to have some additional error checking on distance threshold values depending on the distance unit selected by the user. That is:

1. If `scaled` is selected, the threshold should be >= 0 and <= 100 (the threshold is a percent value).

The complexity of the if/else statements here was because we were passing a list of comma-separated thresholds as a string (e..g, `"1,2,3"`).

For fastmatching, I think it would make sense to keep some of these checks, but they can be simplified (since the threshold is passed as a number instead of a string that needs to be parsed). Specifically:

1. If `hamming` is selected, the threshold is >= 0 (this would already be supported by adding the constraint in the `schema_input.json` file from another of my comments).
2. If `scaled` is selected, the threshold is >= 0 and <= 100 (since it's a percentage value).

What do you think?

Reply:
If I understand correctly `profile_dist` will still output a percentage or integer based on `pd_distm` so the threshold cut-off will vary? I will implement this. I think it makes sense. Could be useful if user forgets to check between scaled and hamming.

Reply:
Operational! a70d030